botocore/docs/sharedexample.py

```diff
@@ -0,0 +1,166 @@
+        self._document_dict(output_section, params, comments, [], True)
+        closing_section = output_section.add_new_section('output-close')
+        closing_section.style.end_codeblock()
+
+    def _document(self, section, value, comments, path):
+        if isinstance(value, dict):
+            self._document_dict(section, value, comments, path)
+        if isinstance(value, list):
+            self._document_list(section, value, comments, path)
+        if isinstance(value, string_types):
+            self._document_str(section, value, comments, path)
+
+    def _document_dict(self, section, value, comments, path, top_level=False):
+        comment = self._get_comment(path, comments)
+        dict_section = section.add_new_section('dict-value')
+        self._start_nested_value(dict_section, '{', comment)
+        for key, val in value.items():
+            path.append('.%s' % key)
+            item_section = dict_section.add_new_section(key)
+            item_section.style.new_line()
+            item_section.write("'%s': " % key)
+            self._document(item_section, val, comments, path)
+            path.pop()
+        dict_section_end = dict_section.add_new_section('ending-brace')
+        self._end_nested_value(dict_section_end, '}')
+        if not top_level:
+            dict_section_end.write(',')
+
+    def _document_params(self, section, value, comments, path):
+        comment = self._get_comment(path, comments)
+        param_section = section.add_new_section('param-values')
+        self._start_nested_value(param_section, '(', comment)
+        for key, val in value.items():
+            path.append('.%s' % key)
+            item_section = param_section.add_new_section(key)
+            item_section.style.new_line()
+            item_section.write(key + '=')
+            self._document(item_section, val, comments, path)
+            path.pop()
+        param_section_end = param_section.add_new_section('ending-parenthesis')
+        self._end_nested_value(param_section_end, ')')
+
+    def _document_list(self, section, value, comments, path):
+        comment = self._get_comment(path, comments)
+        list_section = section.add_new_section('list-section')
+        self._start_nested_value(list_section, '[', comment)
+        for index, val in enumerate(value):
+            item_section = list_section.add_new_section(index)
+            item_section.style.new_line()
+            path.append('[%s]' % index)
+            self._document(item_section, val, comments, path)
+            path.pop()
+        list_section_end = list_section.add_new_section('ending-bracket')
+        self._end_nested_value(list_section_end, '],')
+
+    def _document_str(self, section, value, comments, path):
+        comment = self._get_comment(path, comments)
+        section.write("'%s',%s" % (value, comment))
+
+    def _get_comment(self, path, comments):
+        key = re.sub('^\.', '', ''.join(path))
```

Comment:
I was curious on what the reason for the path is for because it is passed around for all of the methods. May be good to add comments in the code to explain why you are appending to the list.